src/api/migraphx.py

```diff
@@ -220,6 +220,27 @@ def module(h):
              returns='migraphx::instruction_ref')
 
 
+@auto_handle()
+def value(h):
+    h.constructor('create')
+    h.method('is_null', returns='bool', const=True)
+    h.method('get_key', returns='const std::string', const=True)
+    cpp_types = ['int64_t', 'uint64_t', 'double', 'std::string', 'bool']
+    vt = ['int64', 'uint64', 'float', 'string', 'bool']
+    for vt, cpp_type in zip(vt, cpp_types):
+        if (vt == 'string'):
+            h.constructor('create_' + vt, api.params(i='const char*'))
+            h.constructor('create_' + vt + '_with_key',
+                          api.params(pkey='const char*', i='const char*'))
+        else:
+            h.constructor('create_' + vt, api.params(i=cpp_type))
+            h.constructor('create_' + vt + '_with_key',
+                          api.params(pkey='const char*', i=cpp_type))
+        h.method('if_' + vt, returns='const ' + cpp_type + '*', const=True)
+        h.method('is_' + vt, returns='bool', const=True)
+        h.method('get_' + vt, returns=cpp_type, const=True)
```

Comment:
Actually, I dont think we should add the get methods. Instead lets just add `to_string`, `to_int64`, `to_uint64`, and `to_double` method by calling `v.to<T>()`. Even for now, you could just add `to_string`, which should be enough for unit tests.

Reply:
`to<T>()` does the implicit conversion, e.g. `int` can be casted to `float`.  It is not same as `get` methods which has type guarantee.  I am not sure if we want that.

Reply:
> to<T>() does the implicit conversion, e.g. int can be casted to float.

But during deserialization we would do this conversion anyways.

>  It is not same as get methods which has type guarantee. I am not sure if we want that in place of get

By having the exact type, then we need to write visitors for deserializations.

It would be simpler to just have a `.to<T>` method without visitors and such. By exposing, conversions to float, string, signed and unsigned ints, then we can efficiently implement `to<T>` in the C++ API(dont worry about implementing it for this PR though).

Reply:
> then we need to write visitors

This might be a better choice since we dont have to deal with temporary strings.